lms/models/application_instance.py

```diff
@@ -87,15 +87,19 @@ class Product(str, Enum):
     #: (each token belongs to a different user of this application
     #: instance's LMS).
     access_tokens = sa.orm.relationship(
-        "OAuth2Token", back_populates="application_instance"
+        "OAuth2Token",
+        back_populates="application_instance",
+        foreign_keys="OAuth2Token.consumer_key",
     )
 
     #: A list of all the courses for this application instance.
     courses = sa.orm.relationship("LegacyCourse", back_populates="application_instance")
 
     #: A list of all the GroupInfo's for this application instance.
     group_infos = sa.orm.relationship(
-        "GroupInfo", back_populates="application_instance"
+        "GroupInfo",
+        back_populates="application_instance",
+        foreign_keys="GroupInfo.consumer_key",
```

Comment:
Same, we point application_instances.group_infos to make the queries to use GroupInfo.consumer_key.